core/templates/components/forms/custom-forms-directives/object-editor.directive.ts

```diff
@@ -17,142 +17,303 @@
-          if (!scope.ngModelController) {
-            return;
-          }
-          for (const key of Object.keys(e)) {
-            scope.ngModelController.$setValidity(key, e[key]);
-          }
-          scope.$applyAsync();
-        };
-        if (directiveName) {
-          if (MIGRATED_EDITORS.indexOf(directiveName) >= 0) {
-            if (directiveName === (
-              'list-of-sets-of-translatable-html-content-ids'
-            ) && !scope.initArgs) {
-              throw new Error('\nProvided initArgs: ' + scope.initArgs);
+@Component({
+  selector: 'object-editor',
+  template: '',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => ObjectEditorComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => ObjectEditorComponent),
+      multi: true
+    }
+  ]
+})
+export class ObjectEditorComponent
+implements AfterViewInit, OnChanges, OnDestroy,
+ControlValueAccessor, Validator {
+  private _value: unknown;
+  @Input() alwaysEditable: string;
+  @Input() initArgs: unknown;
+  @Input() isEditable: string;
+  @Input() modalId: symbol;
+  @Input() objType: string;
+  @Input() schema;
+  @Input() form;
+  @Output() validityChange: EventEmitter<void> = new EventEmitter();
+  get value(): unknown {
+    return this._value;
+  }
+
+  @Input() set value(val: unknown) {
+    this._value = val;
+    // Ng-model can call write-obj before we create the component. Hence a
+    // check to see if component has been created.
+    if (this.componentRef) {
+      this.componentRef.instance.value = this._value;
+      this.onChange(this._value);
+      this.valueChange.emit(this._value);
+    }
+  }
+
+  @Output() valueChange = new EventEmitter();
+  componentRef: ComponentRef<ObjectEditor>;
+  componentSubscriptions = new Subscription();
+  onChange: (_: unknown) => void = () => {};
+  onTouch: () => void;
+  onValidatorChange: () => void = () => {};
+
+
+  // A hashmap is used instead of an array for faster lookup.
+  componentErrors: Record<string, false> = {};
+
+  getComponentValidationState(): Record<string, boolean> {
+    return this.componentErrors;
+  }
+
+
+  registerOnTouched(fn: () => void): void {
+    this.onTouch = fn;
+  }
+
+  registerOnValidatorChange?(fn: () => void): void {
+    this.onValidatorChange = fn;
+  }
+
+  writeValue(obj: unknown): void {
+    if (obj === null || obj === undefined) {
+      return;
+    }
+    this.value = obj;
+  }
+
+  registerOnChange(fn: (_: unknown) => void): void {
+    this.onChange = fn;
+  }
+
+  constructor(
+    private loggerService: LoggerService,
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private viewContainerRef: ViewContainerRef
+  ) { }
+
+  ngAfterViewInit(): void {
+    const editorName = this.objType.replace(
+      /([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+    if (editorName === (
+      'list-of-sets-of-translatable-html-content-ids'
```

Comment:
This seems odd. Why is this special-cased? There is no explanation, and this sort of if clause shouldn't be in what looks like a generic component.

Ditto below.

Reply:
Yeah. I don't know who added it but it was introduced for debugging purposes. I kept it. We can figure out who added it from the blame but I kept it as it is as to not remove someones debugging efforts.

Reply:
This is in develop. I have kept it as it is.